### PR TITLE
set cookie by domain

### PR DESCRIPTION
**Pull request by cschucode**

### CODE REVIEW

File: skins/oasis/js/privacyUpdateNotification.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ $(function() {
         var notification = new BannerNotification(message).show();
 
         notification.onClose(function() {
-            window.Wikia.Cookies.set('dismissed-privacy-notification', true);
+            window.Wikia.Cookies.set('dismissed-privacy-notification', true, {domain: 'fandom.com'});
```
Comment (vforge) [inline skins/oasis/js/privacyUpdateNotification.js]: I think this should be `{domain: '.fandom.com', path: '/'}` instead.

Comment (JoshuaRogan) [reply skins/oasis/js/privacyUpdateNotification.js]: I agree with V but double check the GDPR lib and use that logic

Comment (cschucode) [reply skins/oasis/js/privacyUpdateNotification.js]: Added path, but don't need the leading dot.

> Contrary to earlier specifications, leading dots in domain names (.example.com) are ignored.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

File: skins/oasis/js/privacyUpdateNotification.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ $(function() {
         var notification = new BannerNotification(message).show();
 
         notification.onClose(function() {
-            window.Wikia.Cookies.set('dismissed-privacy-notification', true);
+            window.Wikia.Cookies.set('dismissed-privacy-notification', true, { domain: 'fandom.com', path: '/' });
```
Comment (mklucsarits) [inline skins/oasis/js/privacyUpdateNotification.js]: The domain needs to be `.fandom.com` with the preceding dot.

File: skins/oasis/js/privacyUpdateNotification.js [hunk 3]
```diff
@@ -4,8 +4,30 @@ $(function() {
         // display notification
         var notification = new BannerNotification(message).show();
 
+        function getCookieDomain(hostname) {
+            const parts = hostname.split('.');
+            if (parts.length < 2) {
```
Comment (mklucsarits) [inline skins/oasis/js/privacyUpdateNotification.js]: This won't happen on our domains, plus if it did it your code would set a cookie on `undefined`, which would throw an error. I don't think this check is needed.

File: skins/oasis/js/privacyUpdateNotification.js [hunk 4]
```diff
@@ -4,8 +4,30 @@ $(function() {
...
+            // These exceptions require a third part for a valid cookie domain. This isn't
+            // a definitive list but rather the most likely domains on which Fandom would
+            // host a site.
+            const exceptions = [
+                '.co.jp',
+                '.co.nz',
+                '.co.uk',
+            ];
+            if (exceptions.indexOf(cookieDomain) >= 0) {
+                cookieDomain = `.${parts[parts.length - 3]}${cookieDomain}`;
```
Comment (JoshuaRogan) [inline skins/oasis/js/privacyUpdateNotification.js]: Is ``` template literals supported on app?

Comment (KockaAdmiralac) [reply skins/oasis/js/privacyUpdateNotification.js]: They shouldn't be, ResourceLoader does not accept accept code that is syntactically invalid in the ES3 standard.

Comment (cschucode) [reply skins/oasis/js/privacyUpdateNotification.js]: Just caught that as well.

**Review by vforge — APPROVED** [review]

**Pull request merged**